scripts/common.py

```diff
@@ -712,3 +715,91 @@ def __enter__(self):
 
     def __exit__(self, etype, value, traceback):
         os.chdir(self.saved_path)
+
+
+@contextlib.contextmanager
+def managed_process(args, shell=False, **kwargs):
+    """Context manager for starting and stopping a process gracefully.
+
+    Args:
+        args: list(*). A sequence of program arguments, where the program to
+            execute is the first item. All items are converted to strings first.
+        shell: bool. Whether the command should be run inside of its own shell.
+        **kwargs: dict(str: *). Same kwargs as `subprocess.Popen`.
+
+    Yields:
+        psutil.Process. The process managed by the context manager.
+    """
+    if PSUTIL_DIR not in sys.path:
+        sys.path.insert(1, PSUTIL_DIR)
+    import psutil
```

Comment:
Why we cannot import this as any other library?

Reply:
Hmm OK, I'll look into fixing this in a separate PR since it's unrelated, but I do want to fix it because I really don't like this pattern.

Reply:
Can you add a TODO for this, and an issue that it points to? We should do this as a matter of course for anything that's intended to be done in the future.

Reply:
Done: #11549